array.c

```diff
@@ -5697,14 +5707,14 @@ rb_ary_eql(VALUE ary1, VALUE ary2)
 
 /*
  *  call-seq:
- *     ary.hash   -> integer
- *
- *  Compute a hash-code for this array.
+ *    array.hash -> integer
  *
- *  Two arrays with the same content will have the same hash code (and will
- *  compare using #eql?).
+ *  Returns the integer hash value for +self+:
+ *    [].hash.class # => Integer
  *
- *  See also Object#hash.
+ *  Two arrays have the same hash value if and only if they have the same content:
```

Comment:
We should mention that the `same content` refers to `eql?` equality, not `==` equality, as the previous documentation did.

Reply:
I still think we should mention `eql?` vs. `==` here, but it's not a big deal.